src/mesh/index_derivs.cxx

```diff
@@ -1278,6 +1499,7 @@ const Field3D Mesh::applyYdiff(const Field3D &var, Mesh::deriv_func func, Mesh::
   result.bndry_xin = result.bndry_xout = result.bndry_yup = result.bndry_ydown = false;
 #endif
   
+  bindex bx;
   if (mesh->freeboundary_xin && mesh->firstX() && !mesh->periodicX) {
     for (bx.jx=mesh->xstart-1; bx.jx>=0; bx.jx--)
       for (bx.jy=mesh->ystart; bx.jy<=mesh->ystart; bx.jy++)
```

Comment:
This whole freeboundary_* code needs to be removed I think, but is probably another PR. I don't think it's used anywhere, and it significantly complicates the (already over-complicated) index_derivs code. 

Based on a switch in mesh, it applies different numerical methods at the first and last points (hence only jy=ystart). A better way is to apply the free boundary conditions on variables where needed.